tracee-ebpf/engine/engine.go

```diff
@@ -0,0 +1,1250 @@
+package tracee_engine
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+	"github.com/aquasecurity/libbpfgo/helpers"
+	"github.com/aquasecurity/tracee/tracee-ebpf/engine/bucketscache"
+	"github.com/aquasecurity/tracee/tracee-ebpf/engine/config"
+	"github.com/aquasecurity/tracee/tracee-ebpf/engine/consts"
+	"github.com/aquasecurity/tracee/tracee-ebpf/engine/event"
+	"github.com/aquasecurity/tracee/tracee-ebpf/engine/filters"
+	"github.com/aquasecurity/tracee/tracee-ebpf/engine/stats"
+	"github.com/aquasecurity/tracee/tracee-ebpf/engine/streamers"
+)
+
+type TraceeEngine interface {
+	Consume() (<-chan event.Event, error)
```

Comment:
I understand what you mean, sounds good however I believe that there are some problems. According to your proposal the consumer is the owner of the channel. However, the ownership on the channel should be on the producer (indeed who writes on the channel should be responsible to close it). In the design you proposed we could end up with a panic easily (for example the consumer decides to close the channel but the producer is not done writing). 
I believe that we still should add the Consume API, leaving the ownership on the producer. Regarding the concern of adding multiple output to be consumed - this can be done in future by returning a struct/map of channels.